packages/votingV2/src/mappings/votingV2.ts

```diff
@@ -304,21 +461,54 @@ export function handleVoterSlashed(event: VoterSlashed): void {
   let user = getOrCreateUser(event.params.voter);
 
   user.cumulativeSlash = defaultBigDecimal(user.cumulativeSlash).plus(toDecimal(event.params.slashedTokens));
-  user.voterActiveStake = toDecimal(event.params.postActiveStake);
+  user.voterStake = toDecimal(event.params.postActiveStake);
+
+  addStakes(user, event.params.postActiveStake, event.block.timestamp);
 
   user.save();
 }
 
-// event ExecutedUnstake(
-//   address indexed voter,
-//   uint256 tokensSent,
-//   uint256 voterActiveStake,
-//   uint256 voterPendingStake
-// );
+// event ExecutedUnstake(address indexed voter, uint256 tokensSent, uint256 voterStake);
 
 export function handleExecutedUnstake(event: ExecutedUnstake): void {
   let user = getOrCreateUser(event.params.voter);
-  user.voterActiveStake = toDecimal(event.params.voterActiveStake);
-  user.voterPendingStake = toDecimal(event.params.voterPendingStake);
+  user.voterStake = toDecimal(event.params.voterStake);
+
+  addStakes(user, event.params.voterStake, event.block.timestamp);
+
   user.save();
 }
+
+function updateAprs(users: string[], emissionRate: BigInt, cumulativeStake: BigDecimal): void {
```

Comment:
not ideal that we have to do this but it worked out to be a lot cleaner than I was worrying it would be. I think having this as synchronous helps